pkg/controller/backup_sync_controller.go

```diff
@@ -134,6 +136,32 @@ func (c *backupSyncController) run() {
 
 	for _, location := range locations {
 		log := c.logger.WithField("backupLocation", location.Name)
+
+		if backupSyncPeriod, ok := location.Spec.Config[backupSyncPeriodKey]; ok {
+			if backupSyncPeriod == "0" {
+				log.Debug("Backup sync period for this location is set to 0, skipping sync")
+				continue
+			}
+
+			syncDuration, err := time.ParseDuration(backupSyncPeriod)
+			if err != nil {
+				log.WithError(err).Error("Error parsing backup sync period for this location")
+				continue
+			}
+
+			log.Debug("Checking if backups need to be synced at this time for this location")
+			lastSync := location.Status.LastSyncedTime
+			nextSync := lastSync.Add(syncDuration)
+			if time.Now().UTC().Before(nextSync) {
+				c.logger.WithFields(logrus.Fields{
+					"backupLocation": location.Name,
+					"lastSync":       lastSync,
+					"nextSync":       nextSync,
+				}).Debug("Sync not required at this time")
+				continue
+			}
```

Comment:
The relationship between this logic and the `--backup-sync-period` could produce some unintuitive behavior in certain instances. For example, if `--backup-sync-period` is set to `1h`, but a specific BSL has a value of `1m`, that BSL will still only get synced every `1h`, because the `run()` function that this check is in will only be run every hour.

I would like to do a little more thinking on how this should work. One option would be to have the `run()` function run frequently (i.e. every 30s? minute? TBD), and each time it runs, check each BSL to see if it should be synced, based on its last-synced time, and either its specific sync interval, or the default value (as provided via `--backup-sync-period`). 

I am also curious - are you mainly interested in being able to disable sync for a specific BSL, or is the custom interval equally important? What's the use case for the custom interval?

Reply:
Agreed, we should change this to the approach proposed by @skriss. The CronJob controller checks [every 10 seconds](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/cronjob/cronjob_controller.go#L98), but I think 30s/1m is also reasonable for this usecase.

Reply:
@skriss @prydonius I updated the PR as the approach suggested, please take a look and let me know any edits I need to make. Sorry for the late update I was OOO towards end of last week.

Reply:
no worries - I'll take a look soon.